### ISSUE TITLE
How do I get a list of already connected BLE devices on my Win10?

### ISSUE THREAD

**Issue by Ugobyte** [inthehand/32feet#323]
Hello,

I am using VS2022, my application is .NET6.0 and I want to get a list of all the BLE devices that are already connected to my Windows 10 PC. These BLE devices are mice, keyboard and Gamepads.

My code looks like this:
![image_2023-07-06_134152987](https://github.com/inthehand/32feet/assets/13201425/6ba3d1c5-844f-4e26-87ff-df10e8911bd1)

However, it never finds any BLE device. When I debug my code, I get this:
![image_2023-07-06_134252079](https://github.com/inthehand/32feet/assets/13201425/2da6eb09-14ff-44e8-bb5f-29f245cee590)

What am I doing wrong? Can someone help point me in the right direction? Any code samples will be greatly appreciated.

**Comment by Ugobyte** [comment]
I was able to get it to work. However, the search only returns my neighbors Samsung TV, hahaha.
I want the search to include BLE devices that are ALREADY connected to my Win10 PC.

The image below shows I have a Bluetooth Touch Mouse and a Bluetooth Gamepad already connected, how can I find the Touch Mouse and Gamepad using inthehand? My goal is to make a Windows app that allows me communicate with an ALREADY connected/paired device.
![image_2023-07-06_151352955](https://github.com/inthehand/32feet/assets/13201425/76e1d061-1dd9-4c74-b663-1da30267031d)

**Comment by peterfoot** [comment]
BluetoothClient is part of the Bluetooth Classic API. The separate InTheHand.BluetoothLE contains Bluetooth LE functionality.

**Comment by Ugobyte** [comment]
> InTheHand.BluetoothLE

Thanks Peter.
I've gone ahead an downloaded the inthehand.BluetoothLE package.
Can you point me where I can see example code or some documentation? I've looked all over but I can't find a thing.

**Comment by peterfoot** [comment]
There is a very rough sample here - [Samples/BluetoothClientApp](https://github.com/inthehand/32feet/tree/main/Samples/BluetoothClientApp)

The API documentation is here - https://inthehand.github.io/html/N_InTheHand_Bluetooth.htm

**Comment by Ugobyte** [comment]
> There is a very rough sample here - [Samples/BluetoothClientApp](https://github.com/inthehand/32feet/tree/main/Samples/BluetoothClientApp)
> 
> The API documentation is here - https://inthehand.github.io/html/N_InTheHand_Bluetooth.htm

Pardon me for asking such questions.
But when I call  `Bluetooth.ScanForDevicesAsync()` i get a **System.PlatformNotSupportedException: 'Operation is not supported on this platform.'** 

I'm on a Win10 PC, using VS2022 with .NET6.0 and I'm trying to make a ClassLibrary project (I'm writing a custom .dll)
Am i getting this exception due to the fact that i writing a .dll? and not a WPF or Windows form project?

Also, I see in the remarks where you guys mentioned **" Windows - Add the Bluetooth permission"** 
Perhaps I'm getting this exception because I did not add Bluetooth permission.
If so, how do i "add" Bluetooth permission?

Thanks Peter, you've been great thus far.

**Comment by peterfoot** [comment]
It sounds like the wrong DLL is being deployed. Does the consuming app target net6.0-windows10.x or higher?

**Comment by Ugobyte** [comment]
> 

Yes.
![image_2023-07-08_131011099](https://github.com/inthehand/32feet/assets/13201425/67fe35a6-7fc4-46c2-92cb-0390a111ac40)

**Comment by Ugobyte** [comment]
![image_2023-07-08_135515555](https://github.com/inthehand/32feet/assets/13201425/feacd217-08d0-4e95-9d4a-5aa0078dde51)

**Comment by Ugobyte** [comment]
I have the latest InTheHand.BluetoothLE which is version 4.0.33. And my target platform is net-3.0-windows10.017763.0. 
Are these incompatible ?

**Comment by Ugobyte** [comment]
@peterfoot Do you have any ideas? or suggestions?

**Comment by berkut0** [comment]
@Ugobyte confirm
Same error when trying to call ScanForDevicesAsync with BLE

**Comment by Ugobyte** [comment]
@berkut0 I will try it shortly and report back.

**Comment by Ugobyte** [comment]
> @Ugobyte confirm Same error when trying to call ScanForDevicesAsync with BLE

Yes i can confirm I'm using .ScanForDevicesAsync. Here is my code 
![image](https://github.com/inthehand/32feet/assets/13201425/a987dd9a-7bb1-46b7-b353-96548125c4cb)
.
.
.
.
.
When I run, I get this error.
![image](https://github.com/inthehand/32feet/assets/13201425/433859ae-e18b-4da3-b599-8ce8e77fc301)
